include/databaseFunctions_blog_pic.php

```diff
@@ -1,58 +1,250 @@
 
 <?php
 
-	function home(){
-		echo"<a href='index.php'>back to home</a>";
+	//RANDOM FUNCTIONS
+	function Home(){
+		echo"
+			<a class='home' href='index.php'>back to home</a>
+			";
+	}
+
+	function ShowDelete($postID){
+		if(isset($_REQUEST['delete'])){
+			if(ValidDelete($postID)){
+				DeletePost($postID);
+				header("Location: index.php");
+			}else{
+				echo"You do not have permission to delete this post.";
+			}
+		}
+		echo"
+			<form method='post'>
+				<input type='submit' name='delete' value='Delete Post'><br><br>
+			</form>
+		";
+
+	}
+
+	function ShowLoginPage(){
+		echo"
+			<a href='login.php'>Log In</a><br>
+		";
+	}
+
+	function CreateAccountPage(){
+		echo"
+			<a href='createAccount.php'>Create an account</a><br>
+		";
+	}
+
+
+	//FORM FUNCTIONS
+	function ValidateTextField($key, $errors){
+		if(!$_REQUEST[$key]){
+			$errors[$key] = "required";
+		}
+		return $errors;
+	}
+
+	function ShowTextField($isreq, $name){
+		echo"
+			<p";
+		if($isreq){
+			echo" class='required'";
+		}
+		echo">$name";
+		if($isreq){
+			echo"*";
+		}
+		echo":</p><input type='text' name='$name'";
+		if(isset($_REQUEST[$name])){
+			echo"value='$_REQUEST[$name]'";
+		}
+		echo"><br><br>
+		";
 	}
 
-	function showDelete($postID){
-		$post = GetPost($postID);
+	function ShowPasswordField($name, $text){
 		echo"
-			<button onClick='scriptDelete($post[delKey])'>Delete Post</button><br><br>
+			<p class='required'>$text*:</p><input type='password' name='$name'";
+		if(isset($_REQUEST[$name])){
+			echo"value='$_REQUEST[$name]'";
+		}
+		echo"><br><br>
 		";
+	}
+
+	//USER FUNCTIONS
+	function CreateAccount(){
+		$errors = array();
+		if(isset($_REQUEST['create'])){
+			$errors+=ValidateTextField('username', $errors);
+			$errors+=ValidateTextField('password', $errors);
+			$errors+=ValidateTextField('confirm', $errors);
+			$errors+=ValidateTextField('email', $errors);
+			if((!isset($errors['password']))&&(!isset($errors['confirm']))){
+				if($_REQUEST['password'] != $_REQUEST['confirm']){
+					$errors['match'] = 'error';
+				}
+			}
+			if(sizeof($errors) == 0){
+				AddNewUser($_REQUEST['username'], $_REQUEST['password'], $_REQUEST['email']);
+				header("Location: login.php");
+			}else{
+				if(isset($errors['username'])){
```

Comment:
You've got a bunch of if statements here that are all very similar. Can you think of a way to expand your form functions to make this a bit cleaner and more automated?